.github/workflows/run_periodic_tests.yml

```diff
@@ -78,98 +77,29 @@ jobs:
-
-      - name: Set up Python 3.12
-        uses: actions/setup-python@v5
-        with:
-          python-version: 3.12
-
-      - name: Install nox
-        run: python -m pip install nox
-
-      - name: Install SuiteSparse and SUNDIALS on GNU/Linux
-        timeout-minutes: 10
-        run: python -m nox -s pybamm-requires
-
-      - name: Run unit tests for Ubuntu with Python 3.12 and generate coverage report
+      - name: Run coverage tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.12'
         run: python -m nox -s coverage
 
       - name: Upload coverage report
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.12'
         uses: codecov/codecov-action@v4.5.0
-        if: github.repository == 'pybamm-team/PyBaMM'
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
-  run_integration_tests:
-    name: Integration tests (${{ matrix.os }} / Python ${{ matrix.python-version }})
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, macos-12, macos-14, windows-latest]
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
-    steps:
-      - name: Check out PyBaMM repository
-        uses: actions/checkout@v4
-
-      - name: Install Linux system dependencies
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          sudo apt-get update
-          sudo apt-get install gfortran gcc graphviz pandoc libopenblas-dev texlive-latex-extra dvipng
-
-      - name: Install macOS system dependencies
-        if: matrix.os == 'macos-12' || matrix.os == 'macos-14'
-        env:
-          HOMEBREW_NO_INSTALL_CLEANUP: 1
-          HOMEBREW_NO_AUTO_UPDATE: 1
-          HOMEBREW_NO_COLOR: 1
-          # Speed up CI
-          NONINTERACTIVE: 1
-        # sometimes gfortran cannot be found, so reinstall gcc just to be sure
-        run: |
-          brew analytics off
-          brew install graphviz
-          brew reinstall gcc
-
-      - name: Install Windows system dependencies
-        if: matrix.os == 'windows-latest'
-        run: choco install graphviz --version=8.0.5
-
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-
-      - name: Install nox
-        run: python -m pip install nox
-
-      - name: Install SuiteSparse and SUNDIALS on GNU/Linux and macOS
-        timeout-minutes: 10
-        if: matrix.os != 'windows-latest'
-        run: python -m nox -s pybamm-requires
-
       - name: Run integration tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
         run: python -m nox -s integration
 
   # Skips IDAKLU module compilation for speedups, which is already tested in other jobs.
   run_doctests:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```

Comment:
```suggestion
```

This is a single job and not a matrix and therefore does not require a strategy definition